src/VTMap.js

```diff
@@ -0,0 +1,371 @@
+import React from 'react';
+import './App.css';
+import { Map, TileLayer, Polygon, Polyline } from 'react-leaflet';
+import borderData from './border.js';
+import L from 'leaflet'
+import leafletPip from 'leaflet-pip'
+let defaultZoom = 7
+
+class VTMap extends React.Component {
+    constructor(props) {
+        super(props)
+
+        this.state = {
+            latitude: '',
+            longitude: '',
+            county: '',
+            town: '',
+            status: 'Game Started, Awaiting your first guess..',
+            guess: '',
+            modalDisplayed: false,
+            points: 100,
+            mapLat: 44.0886,
+            mapLon: -72.7317,
+            locationArray: [],
+            highScore: '',
+            username: 'Unknown'
+            
+        }
+    }
+
+    // itterate over indexs [0]  and [1] in each array in the list of arrays and return the highest of each value
+    getCorners = () => {
+
+        let maxLon = -73.35218
+        let minLon = -71.510225
+
+        let maxLat = 45.005419
+        let minLat = 42.730315
+        return (
+            [minLat, maxLat, minLon, maxLon]
+        )
+    }
+
+    getRndInteger = (min, max) => {
+        return Math.floor(Math.random() * (max - min)) + min;
+    }
+
+    randPoint = (array) => {
+        let randoLon = this.getRndInteger(array[0], array[1])
+
+        let randoLat = this.getRndInteger(array[2], array[3])
+
+        return (
+            [randoLon, randoLat]
+
+        )
+    }
+
+    // Logic to check if the randomly chosen point is within the VTBorder
+    checkValidPoint = (layer, rect, pin) => {
+        if (layer.length) {
+            console.log(`Correct length, returning pin: ${pin}`)
+            return pin
+        }
+        else {
+            console.log("Invalid point...")
+            let newPin = this.randPoint(rect)
+            let newLayer = (leafletPip.pointInLayer([newPin[1], newPin[0]], L.geoJSON(borderData)))
+            return this.checkValidPoint(newLayer, rect, newPin)
+        }
+    }
+
+    clickHandlerStart = (evt) => {
+        evt.preventDefault();
+
+        this.setState({
+            playing: true,
+            points: 100
+        })
+        // let myMap = document.getElementById('mainMap')
+
+        let corners = this.getCorners(borderData)
+
+        let pinDraft = this.randPoint(corners) //this is the point on the map we are going to check
+
+        let geoJsonData = L.geoJSON(borderData)
+
+        let layerDraft = (leafletPip.pointInLayer([pinDraft[0], pinDraft[1]], geoJsonData))
+
+        let validPoint = this.checkValidPoint(layerDraft, corners, pinDraft)
+
+        const {locationArray} = this.state
+
+        const latLonArray = locationArray.concat([validPoint])
+
+        console.log(`This is the valid point: ${validPoint}`)
+        this.setState({
+            mapLat: validPoint[0],
+            mapLon: validPoint[1],
+            locationArray: latLonArray
+        })
+
```

Comment:
Glad we got this working! The problem is, it persists across playthroughs! Your breadcrumbs are not reset after the player gives up, so the `Polyline` component just draws more and more with an ever-increasing `latLonArray`.  Ensure you `setState` when needed to "clean up" after a play through!